File: HelloDockerWin/ClientApp/app/shared/city.ts
export class CityToAdd {
    name: string;
    description: string;
}

export class City extends CityToAdd {
    id: string;
    pointsOfInterest: PointOfInterest[] = new Array<PointOfInterest>();
}

export class PointOfInterest {
    id: string;
    name: string;
    description: string;
}

File: HelloDockerWin/Entities/City.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HelloDockerWin.Entities
{
    public class City
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; }

        [MaxLength(200)]
        public string Description { get; set; }
        public ICollection<PointOfInterest> PointsOfInterest { get; set; }
                = new List<PointOfInterest>();

        public Principal Principal { get; set; }    
    }
}
